### PR TITLE
Update export/write services for eventual tensor storage

### PR DESCRIPTION
**Pull request by ericdnielsen**
Adds fields to experiment, and subsidiary messages for tensor storage issues.  Adds the placeholder request/response.  I need these definitions to exist for other dependent changes, and hope we can review these in isolation.

This is not adding implementation of these methods.

We may? want to use oneof for the Tag level Scalars or Tensors points

### CODE REVIEW

**Comment by ericdnielsen** [comment]
I know this will have merge issues with #3234

**Comment by ericdnielsen** [comment]
Found some issues, removed reviewers while I fix it.

**Comment by ericdnielsen** [comment]
Resolved merge issues post Experiment name/description PR.

File: tensorboard/uploader/proto/export_service.proto [hunk 1]
```diff
@@ -154,4 +161,17 @@ message StreamExperimentDataResponse {
     // Value of the point at this step / timestamp.
     repeated double values = 3;
   }
+
+  // Data for the Tensors are stored in a columnar fashion to optimize it for
+  // exporting the data into textual formats like JSON.
+  // The data for the ith scalar is { steps[i], wall_times[i], values[i] }.
```
Comment (davidsoergel) [inline tensorboard/uploader/proto/export_service.proto]: tensor

File: tensorboard/uploader/proto/tensor.proto [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  // Step index within the run.
+  int64 step = 1;
+  // Timestamp of the creation of this point.
+  google.protobuf.Timestamp wall_time = 2;
+  // Value of the point at this step / timestamp.
+  .tensorboard.TensorProto value = 3;
+}
+
+// Metadata for the TensorPoints stored for one (Experiment, Run, Tag).
+message TensorPointMetadata {
```
Comment (davidsoergel) [inline tensorboard/uploader/proto/tensor.proto]: Is this used anywhere?  It appears forked from ScalarPointMetadata, which is also unused?

File: tensorboard/uploader/proto/export_service.proto (hunk 1 above)
Comment (ericdnielsen) [reply tensorboard/uploader/proto/export_service.proto]: done

File: tensorboard/uploader/proto/tensor.proto (hunk 2 above)
Comment (ericdnielsen) [reply tensorboard/uploader/proto/tensor.proto]: Its used in the storage backend, but defined here.

**Pull request merged**